packages/ketchup/src/components/kup-activity-timeline/kup-activity-timeline.tsx

```diff
@@ -0,0 +1,393 @@
+    KupComponent,
+    KupEventPayload,
+} from '../../types/GenericTypes';
+import { getProps, setProps } from '../../utils/utils';
+import {
+    ActivityTimeline,
+    KupActivity,
+    KupActivityTimelineAction,
+    KupActivityTimelineData,
+    KupActivityTimelineProps,
+    KupDatatableClickEventPayload,
+} from './kup-activity-timeline-declarations';
+import { FImage } from '../../f-components/f-image/f-image';
+import { getCellValueForDisplay } from '../../utils/cell-utils';
+
+@Component({
+    tag: 'kup-activity-timeline',
+    styleUrl: 'kup-activity-timeline.scss',
+    shadow: true,
+})
+export class KupActivityTimeline {
+    /**
+     * References the root HTML element of the component (<kup-activity-timeline>).
+     */
+    @Element() rootElement: HTMLElement;
+
+    /*-------------------------------------------------*/
+    /*                   S t a t e s                   */
+    /*-------------------------------------------------*/
+
+    @State()
+    timeline: ActivityTimeline[] = [];
+
+    /*-------------------------------------------------*/
+    /*                    P r o p s                    */
+    /*-------------------------------------------------*/
+
+    /**
+     *  Dataset containing the activities list.
+     * @default null
+     */
+    @Prop() data: KupDataDataset;
+
+    /**
+     *  Columns containing dates.
+     * @default null
+     */
+    @Prop() dateColumn: string;
+
+    /**
+     *  Columns containing times.
+     * @default null
+     */
+    @Prop() timeColumn: string;
+    /*-------------------------------------------------*/
+    /*       I n t e r n a l   V a r i a b l e s       */
+    /*-------------------------------------------------*/
+
+    #kupManager: KupManager = kupManagerInstance();
+
+    /*-------------------------------------------------*/
+    /*                   E v e n t s                   */
+    /*-------------------------------------------------*/
+
+    /**
+     * Generic click event on activity.
+     */
+    @Event({
+        eventName: 'kup-activity-timeline-click',
+        composed: true,
+        cancelable: false,
+        bubbles: true,
+    })
+    kupClick: EventEmitter<KupDatatableClickEventPayload>;
+
+    /**
+     * Generic right click event on activity.
+     */
+    @Event({
+        eventName: 'kup-activity-timeline-contextmenu',
+        composed: true,
+        cancelable: false,
+        bubbles: true,
+    })
+    kupContextMenu: EventEmitter<KupDatatableClickEventPayload>;
+
+    /**
+     * Triggered when the component is ready.
+     */
+    @Event({
+        eventName: 'kup-activity-timeline-ready',
+        composed: true,
+        cancelable: false,
+        bubbles: true,
+    })
+    kupReady: EventEmitter<KupEventPayload>;
+
+    /*-------------------------------------------------*/
+    /*                  W a t c h e r s                */
+    /*-------------------------------------------------*/
+
+    @Watch('data')
```

Comment:
Why the need to use the `Watch` decorator?
Changing a prop triggers a new rendering cycle: it would be better to seamlessly integrate the creation of the widget in the rendering process.
By looking at the #toTimeline method I can't spot the need to use `Watch`, maybe I'm missing something?